src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/SetAzureKeyVaultManagedStorageAccount.cs

```diff
@@ -0,0 +1,101 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.KeyVault.Models;
+using System;
+using System.Collections;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.KeyVault
+{
+    [Cmdlet( VerbsCommon.Set, CmdletNoun.AzureKeyVaultManagedStorageAccount,
+        SupportsShouldProcess = true,
+        HelpUri = Constants.KeyVaultHelpUri)]
+    [OutputType(typeof(ManagedStorageAccount))]
+    public class SetAzureKeyVaultManagedStorageAccount : KeyVaultCmdletBase
+    {
+        #region Input Parameter Definitions
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
Reiterating the pieplien note form previously (this applies to all cmdlets).

Our advice is for piping within a module, you should *not* use ValueFromPipelineByPropertyName, and use ValueFromPipeline with a new parameter set that allows piping the output type from the cmdlet that creates the type you are piping.

i.e.:

```powershell
[Parameter(Mandatory=true, ValueFromPipeline=true, ParameterSetName="KeyVaultPipingParameterSet")]
public KeyVaultKey InputObject {get; set;}
```

Note that this is not a *required* change, but a suggested change